x/evm/types/params.go

```diff
@@ -61,43 +58,43 @@ func NewParams(evmDenom string, enableCreate, enableCall bool, config ChainConfi
 func DefaultParams() Params {
 	return Params{
 		EvmDenom:            DefaultEVMDenom,
-		EnableCreate:        true,
-		EnableCall:          true,
+		EnableCreate:        DefaultEnableCreate,
+		EnableCall:          DefaultEnableCall,
 		ChainConfig:         DefaultChainConfig(),
-		ExtraEIPs:           nil,
+		ExtraEIPs:           DefaultExtraEIPs,
 		AllowUnprotectedTxs: DefaultAllowUnprotectedTxs,
 	}
 }
 
-// ParamSetPairs returns the parameter set pairs.
-func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
-	return paramtypes.ParamSetPairs{
-		paramtypes.NewParamSetPair(ParamStoreKeyEVMDenom, &p.EvmDenom, validateEVMDenom),
-		paramtypes.NewParamSetPair(ParamStoreKeyEnableCreate, &p.EnableCreate, validateBool),
-		paramtypes.NewParamSetPair(ParamStoreKeyEnableCall, &p.EnableCall, validateBool),
-		paramtypes.NewParamSetPair(ParamStoreKeyExtraEIPs, &p.ExtraEIPs, validateEIPs),
-		paramtypes.NewParamSetPair(ParamStoreKeyChainConfig, &p.ChainConfig, validateChainConfig),
-		paramtypes.NewParamSetPair(ParamStoreKeyAllowUnprotectedTxs, &p.AllowUnprotectedTxs, validateBool),
-	}
-}
-
 // Validate performs basic validation on evm parameters.
 func (p Params) Validate() error {
-	if err := sdk.ValidateDenom(p.EvmDenom); err != nil {
+	if err := validateEVMDenom(p.EvmDenom); err != nil {
+		return err
+	}
+
+	if err := validateEIPs(p.ExtraEIPs.ExtraEIPs); err != nil {
+		return err
+	}
+
+	if err := validateBool(p.EnableCall); err != nil {
+		return err
+	}
+
+	if err := validateBool(p.EnableCreate); err != nil {
 		return err
 	}
 
-	if err := validateEIPs(p.ExtraEIPs); err != nil {
+	if err := validateBool(p.AllowUnprotectedTxs); err != nil {
 		return err
 	}
 
-	return p.ChainConfig.Validate()
+	return validateChainConfig(p.ChainConfig)
 }
 
-// EIPs returns the ExtraEips as a int slice
+// EIPs returns the ExtraEIPS as a int slice
 func (p Params) EIPs() []int {
-	eips := make([]int, len(p.ExtraEIPs))
-	for i, eip := range p.ExtraEIPs {
+	eips := make([]int, len(p.ExtraEIPs.ExtraEIPs))
+	for i, eip := range p.ExtraEIPs.ExtraEIPs {
```

Comment:
Nit: could we refactor this to avoid repeating `ExtraEIPs`? For example, we could use
```suggestion
	eips := make([]int, len(p.ExtraEIPs.EIPs))
	for i, eip := range p.ExtraEIPs.EIPs {
```